### PR TITLE
Feature/v1-242 | Refactor controller types

### PR DESCRIPTION
**Pull request by dapodev**
# Overview 

Extracted controller types to separate files. + Minor code changes

### CODE REVIEW

File: backend/src/db/providers/courseProvider.ts [hunk 1]
```diff
@@ -152,13 +150,8 @@ const getCourseProvider = async (courseId: string | ObjectId, userId: string | O
...
 };
 
-const getMaterialsProvider = async ({ courseId, stage }: { courseId: string; stage?: string }) => {
-  const material = await CourseModel.findOne(
-    {
-      $and: [{ _id: courseId }, stage?.length ? { 'materials.stage': Number(stage) } : {}],
-    },
-    stage?.length ? { 'materials.$': 1 } : { materials: 1 },
-  ).lean();
+const getMaterialsProvider = async (courseId: string) => {
```
Comment (SaXaPhonist) [inline backend/src/db/providers/courseProvider.ts:153]: how do we get materials for specific stage?

File: backend/src/controllers/courses/getMaterials.ts [hunk 2]
```diff
@@ -1,12 +1,19 @@
...
+const getMaterials = async (
+  req: TGetMaterialsRequest,
+  res: TGetMaterialsResponse,
+  next: NextFunction,
+) => {
   try {
     const { id: courseId } = req.params;
-    const queryParams = req.query;
-    const material = await getMaterialsProvider({ courseId, ...queryParams });
+    const material = await getMaterialsProvider(courseId);
```
Comment (SaXaPhonist) [inline backend/src/controllers/courses/getMaterials.ts:16]: what's about query string ?

File: backend/src/interfaces/requests/clientCourses/applyCourse.d.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import { IAuthLocals } from '../common/locals';
+import { IIdParams } from '../common/params';
+import { IIdPayload } from '../common/payloads';
+
+type TRequestParams = IIdParams;
+type TRequestBody = IIdPayload;
+
+type TResponsePayload = IClientCourse;
+type TResponseLocals = IAuthLocals & {
+  courseId?: string; // refactor after removing adapters
```
Comment (SaXaPhonist) [inline backend/src/interfaces/requests/clientCourses/applyCourse.d.ts:13]: do you need all  comments?

File: backend/src/interfaces/requests/clientCourses/manageAssessment.d.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import { IIdParams } from '../common/params';
+import { IAssessmentActionPayload } from '../common/payloads';
+
+type TRequestParams = IIdParams;
+type TRequestBody = IAssessmentActionPayload;
+
+type TResponsePayload = string;
+type TResponseLocals = IAuthLocals & {
+  achievments: TAchievements;
+  results: TResponsePayload; // remove results in "V1-247 Remove adapters"
```
Comment (SaXaPhonist) [inline backend/src/interfaces/requests/clientCourses/manageAssessment.d.ts:14]: //

File: backend/src/interfaces/requests/clientCourses/startCourse.d.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { TBaseRequest, TBaseResponse } from '../base';
+import { IIdParams } from '../common/params';
+
+type TRequestParams = IIdParams;
+
+type TResponsePayload = {
+  start: boolean; // refactor in "V1-248 Common response style"
```
Comment (SaXaPhonist) [inline backend/src/interfaces/requests/clientCourses/startCourse.d.ts:7]: //

File: backend/src/interfaces/requests/courses/getMaterials.d.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import { ICourse } from 'interfaces/Ientities/Icourses';
+
+import { TBaseRequest, TBaseResponse } from '../base';
+import { IIdParams } from '../common/params';
+
+type TRequestParams = IIdParams;
+
+type TResponsePayload = Pick<ICourse, '_id' | 'materials'>; // actually should return only IMaterial[]
```
Comment (SaXaPhonist) [inline backend/src/interfaces/requests/courses/getMaterials.d.ts:8]: //

File: backend/src/interfaces/requests/courses/getCoursesMap.d.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import { ICoursesMapResponse } from 'interfaces/IResponse/IResponse';
+
+import { TBaseRequest, TBaseResponse } from '../base';
+import { IAuthLocals } from '../common/locals';
+
+type TResponsePayload = ICoursesMapResponse;
+type TResponseLocals = IAuthLocals & { results: TResponsePayload }; // remove results in "V1-247 Remove adapters"
```
Comment (SaXaPhonist) [inline backend/src/interfaces/requests/courses/getCoursesMap.d.ts:7]: //

File: backend/src/interfaces/requests/manager/assignEmployeeCourses.d.ts [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+import { TBaseRequest, TBaseResponse } from '../base';
+import { IAuthLocals } from '../common/locals';
+import { IIdParams } from '../common/params';
+
+type TRequestParams = IIdParams;
+type TRequestBody = ICourseToAssign[];
+
+type TResponsePayload = string;
+type TResponseLocals = IAuthLocals & {
+  results: TResponsePayload; // remove results in "V1-247 Remove adapters"
```
Comment (SaXaPhonist) [inline backend/src/interfaces/requests/manager/assignEmployeeCourses.d.ts:12]: //

File: backend/src/interfaces/requests/manager/getEmployeeAvailableCourses.d.ts [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+import { IIdParams } from '../common/params';
+import { ITitleQuery } from '../common/queries';
+
+type TRequestParams = IIdParams;
+type TRequestBody = never;
+type TRequestQuery = ITitleQuery;
+
+type TResponsePayload = TAvailableCourse[];
+type TResponseLocals = IAuthLocals & {
+  results: TResponsePayload; // remove results in "V1-247 Remove adapters"
```
Comment (SaXaPhonist) [inline backend/src/interfaces/requests/manager/getEmployeeAvailableCourses.d.ts:14]: //

File: backend/src/interfaces/requests/manager/getPendingAssessments.d.ts [hunk 10]
```diff
@@ -0,0 +1,12 @@
+import { TAssessmentRequest } from 'interfaces/IResponse/IResponse';
+
+import { TBaseRequest, TBaseResponse } from '../base';
+import { IAuthLocals } from '../common/locals';
+
+type TResponsePayload = TAssessmentRequest[];
+type TResponseLocals = IAuthLocals & { results: TResponsePayload }; // remove results in "V1-247 Remove adapters"
```
Comment (SaXaPhonist) [inline backend/src/interfaces/requests/manager/getPendingAssessments.d.ts:7]: //

File: backend/src/interfaces/requests/notifications/addNotification.d.ts [hunk 11]
```diff
@@ -0,0 +1,11 @@
+import { TNotification } from 'interfaces/IResponse/IResponse';
+
+import { TBaseRequest, TBaseResponse } from '../base';
+
+type TResponsePayload = never;
+type TResponseLocals = TNotification; // refactor after removing adapters
```
Comment (SaXaPhonist) [inline backend/src/interfaces/requests/notifications/addNotification.d.ts:6]: //

**Review by SaXaPhonist — CHANGES_REQUESTED** [review]

File: backend/src/interfaces/requests/clientCourses/applyCourse.d.ts (hunk 3 above)
Comment (dapodev) [reply backend/src/interfaces/requests/clientCourses/applyCourse.d.ts:13]: yep we need

Comment (dapodev) [reply backend/src/interfaces/requests/clientCourses/applyCourse.d.ts:13]: check tickets in TODO column pls

File: backend/src/db/providers/courseProvider.ts (hunk 1 above)
Comment (dapodev) [reply backend/src/db/providers/courseProvider.ts:153]: Do we use it anywhere?

File: backend/src/controllers/courses/getMaterials.ts (hunk 2 above)
Comment (dapodev) [reply backend/src/controllers/courses/getMaterials.ts:16]: no need to use them anymore

**Review by zhuk-vezunchikz — APPROVED** [review]

**Review by SaXaPhonist — APPROVED** [review]

**Pull request merged**